relayer/src/config.rs

```diff
@@ -262,6 +262,33 @@ impl Default for RestConfig {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(
+    rename_all = "lowercase",
+    tag = "derivation",
+    content = "proto_type",
+    deny_unknown_fields
+)]
+pub enum AddressType {
+    Cosmos,
+    Ethermint { pk_type: String },
```

Comment:
Wondering if it's a good idea to restrict the possible values that `pk_type` can take, either by making this an enum, or checking the String against a predefined list of accepted values.

Beside `'/injective.crypto.v1beta1.ethsecp256k1.PubKey'` and `"/ethermint.crypto.v1alpha1.ethsecp256k1.PubKey"` are there numerous other values that `pk_type` can take?

Reply:
There are potentially many more : `/ethermint.crypto.v1beta1.ethsecp256k1.PubKey`, `/ethermint.crypto.v1.ethsecp256k1.PubKey`, `/cosmos.crypto.ethsecp256k1.PubKey`...

Right now, in order to be useful (given Ethermint has been copied over and modified), it makes sense to keep it a bit flexible -- otherwise, every chain would need to open a PR to modify the enum.

Later, once the ethsecp256k1 support lands in Cosmos SDK, it'll make sense to deprecate this with that Cosmos SDK release and recommend chains to migrate to `/cosmos.crypto.ethsecp256k1.PubKey` (or whatever the canonical proto ends up being)... and after that, remove this extra config.

Reply:
Sounds like a plan, thanks for the detailed rationale! I agree with you.